### PR TITLE
(fix) Authoring: Refactoring the action for authoring.

### PR DESCRIPTION
**Pull request by mdhaman**

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -822,34 +931,36 @@
...
-                        $scope.origItem.lock_user = $scope.item.lock_user = result.lock_user;
-                        $scope.item._locked = result._locked;
-                        $scope.origItem.lock_session = $scope.item.lock_session = result.lock_session;
-                        $scope._editable = $scope.item._editable = true;
-                        $scope.dirty = false;
-                        $scope.item._autosave = null;
-                    }, 200);
-                }
-
+                // function updateEditorState (result) {
```
Comment (amagdas) [inline client/app/scripts/superdesk-authoring/authoring.js]: Should be removed.

**Comment by amagdas** [comment]
:100: :heart: :blue_heart: :yellow_heart: :green_heart:

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 2]
```diff
@@ -31,6 +31,25 @@
...
+        edit: false,
+        mark_item: false,
+        duplicate: false,
+        copy: false,
+        view: true,
+        spike: false,
+        unspike: false,
+        package_item: false,
+        multi_edit: false
+    };
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: It's a good idea to use `Object.freeze()` for such "constant" objects.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -630,7 +739,7 @@
                         $scope.origItem = res;
                         $scope.dirty = false;
                         $scope.item = _.create($scope.origItem);
-                        $scope.not_for_publication_visible = $scope.publish_enabled() && res.marked_for_not_publication === false;
+                        //$scope.not_for_publication_visible = $scope.publish_enabled() && res.marked_for_not_publication === false;
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: If this is not needed anymore, it's probably better to remove it completely IMO.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 4]
```diff
@@ -822,34 +931,36 @@
...
-                    } else {
-                        superdesk.intent($scope.intentFilter.action, $scope.intentFilter.type, $scope.origItem);
-                    }
+                    superdesk.intent($scope.intentFilter.action, $scope.intentFilter.type, $scope.origItem);
+                    // var path = $location.path();
+                    // if (path.indexOf('/view') < 0) {
+                    //     lock.lock($scope.item, true).then(updateEditorState);
+                    // } else {
+                    //     superdesk.intent($scope.intentFilter.action, $scope.intentFilter.type, $scope.origItem);
+                    // }
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Should probably be removed if not needed anymore.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 5]
```diff
@@ -871,11 +982,15 @@
...
-                            authoring.lock($scope.item, data.user);
+                        authoring.lock($scope.item, data.user);
+                        if ($scope.action !== 'view') {
                             $location.url($scope.referrerUrl);
                         }
+                        // var path = $location.path();
+                        // if (path.indexOf('/view') < 0) {
+                        //     authoring.lock($scope.item, data.user);
+                        //     $location.url($scope.referrerUrl);
+                        // }
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Same comment as above (there is a history in case we need the old code snippet back).

**Comment by plamut** [comment]
Unit tests are missing for the Python changes.

The rest is mostly good in general (besides some relatively minor remarks) and it's especially great to see docstring comments and tests for the new client-side code!

**Comment by mdhaman** [comment]
Thanks @plamut. Will add the unit tests.

**Comment by plamut** [comment]
Cool. Oh, and adding :+1: in advance.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 6]
```diff
@@ -328,6 +361,94 @@
 
             return api.save('archive_link', {}, data, item);
         };
+
+        /**
+        * Actions that it can perform on an item
+        * @param {Object} item : item
+        * @param {Object} privileges : user privileges
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Oh, just noticed - `privileges` is not a parameter here, it's a service dependency.

**Pull request merged**